File: app/src/main/java/com/denproj/posmanongjaks/repository/base/ItemRepository.java
package com.denproj.posmanongjaks.repository.base;

import androidx.lifecycle.MutableLiveData;

import com.denproj.posmanongjaks.model.Item;
import com.denproj.posmanongjaks.util.OnFetchFailed;

import java.util.List;

import kotlin.NotImplementedError;

public interface ItemRepository{

    default MutableLiveData<List<Item>> observeItemsFromBranch(String branchId, OnFetchFailed onFetchFailed) {
        throw new NotImplementedError();
    }
}

File: app/src/main/java/com/denproj/posmanongjaks/repository/base/ItemRepository.kt
package com.denproj.posmanongjaks.repository.base

import androidx.lifecycle.LiveData
import com.denproj.posmanongjaks.model.Item
import com.denproj.posmanongjaks.util.OnDataReceived

interface ItemRepository {
    suspend fun observeItemsFromBranch(
        branchId: String,
        onDataFetched: OnDataReceived<List<Item>>
    )

}
